### PR TITLE
python3.pkgs.pytomlpp: init at 0.3.5

### PR DESCRIPTION
**Pull request by evils**
###### Motivation for this change
same as #122400 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
  - `/nix/store/ajc7l7icrmnaabzv4vjfz5aw8fj5s3sr-python3.8-pytomlpp-0.3.5	  104723184`
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122402` at 865f23ad run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/4a7c8430e7605b9b99d279124d99378c">python38Packages.pytomlpp</a></li>
    <li><a href="https://gist.github.com/15e4203bf20a68c280e4fae46fc7a29c">python39Packages.pytomlpp</a></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-imports-check-typo.md">python-imports-check-typo</a>

A likely typo in the `pythonImportCheck` argument was found. Did you mean `pythonImportsCheck`?

Near pkgs/development/python-modules/pytomlpp/default.nix:43:3:
```
   |
43 |   pythonImportCheck = [ "pytomlpp" ];
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by evils** [comment]
doh, i didn't update the hash after removing `fetchSubmodules` to confirm that was needed

File: pkgs/development/python-modules/pytomlpp/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    dateutil
+    doxygen
+    python
+    pelican
+    matplotlib
+  ];
+
+  preCheck = ''
+    cd tests
+  '';
```
Comment (fabaff) [inline pkgs/development/python-modules/pytomlpp/default.nix:42]: Should work:

```suggestion
  pytestFlagsArray = [ "tests/" ];
```

File: pkgs/development/python-modules/pytomlpp/default.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    fetchSubmodules = true;
+    sha256 = "1h06a2r0f5q4mml485113mn7a7585zmhqsk2p1apcybyydllcqda";
+  };
+
+  buildInputs = [ pybind11 ];
+
+  checkInputs = [
+    pytestCheckHook
+
+    dateutil
```
Comment (fabaff) [inline pkgs/development/python-modules/pytomlpp/default.nix:33]:
```suggestion
    pytestCheckHook
    dateutil
```

File: pkgs/development/python-modules/pytomlpp/default.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    cd tests
+  '';
+
+  pythonImportsCheck = [ "pytomlpp" ];
+
+  meta = with lib; {
+    description = "A python wrapper for tomlplusplus";
+    homepage    = "https://github.com/bobfang1992/pytomlpp";
+    license     = licenses.mit;
+    maintainers = with maintainers; [ evils ];
```
Comment (fabaff) [inline pkgs/development/python-modules/pytomlpp/default.nix:50]: A lot of maintainers are using `nixkgs-fmt` to do the formatting. With the first run the style of `meta` would be updated probably if somebody else is working on the package.

Comment (evils) [reply pkgs/development/python-modules/pytomlpp/default.nix:50]: this style was copied from another file i used as a template, will remove it

File: pkgs/development/python-modules/pytomlpp/default.nix (hunk 1 above)
Comment (evils) [reply pkgs/development/python-modules/pytomlpp/default.nix:43]: same as in the rtoml case, specifying `pytestFlagsArray` instead has the same effect as specifying neither

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 122402 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages built:</summary>  <ul>    <li>python38Packages.pytomlpp</li>    <li>python39Packages.pytomlpp</li>  </ul></details>

**Pull request merged**